src/Microsoft.NetCore.Analyzers/UnitTests/Runtime/DoNotUseEnumerableMethodsOnIndexableCollectionsInsteadUseTheCollectionDirectlyTests.Fixer.cs

```diff
@@ -29,5 +30,186 @@ protected override CodeFixProvider GetCSharpCodeFixProvider()
         {
             return new CSharpDoNotUseEnumerableMethodsOnIndexableCollectionsInsteadUseTheCollectionDirectlyFixer();
         }
+
+        [Fact]
+        public void CA1826FixEnumerableFirstExtensionCallCSharp()
+        {
+            VerifyCSharpFix(@"
+using System;
+using System.Linq;
+using System.Collections.Generic;
+class C
+{
+    void M()
+    {
+        var list = GetList();
+        var matrix = new[] { list };
+        var f1 = list.First();
+        var f2 = GetList().First();
+        var f3 = matrix[0].First();
+        Console.WriteLine(list.First());
+    }
+
+    IReadOnlyList<int> GetList()
+    {
+        return new List<int> { 1, 2, 3 };
+    }
+}
+", @"
+using System;
+using System.Linq;
+using System.Collections.Generic;
+class C
+{
+    void M()
+    {
+        var list = GetList();
+        var matrix = new[] { list };
+        var f1 = list[0];
+        var f2 = GetList()[0];
+        var f3 = matrix[0][0];
+        Console.WriteLine(list[0]);
+    }
+
+    IReadOnlyList<int> GetList()
+    {
+        return new List<int> { 1, 2, 3 };
+    }
+}
+", allowNewCompilerDiagnostics: true);
+        }
+
+        [Fact]
+        public void CA1826FixEnumerableFirstStaticCallCSharp()
+        {
+            VerifyCSharpFix(@"
+using System;
+using System.Linq;
+using System.Collections.Generic;
+class C
+{
+    void M()
+    {
+        var list = GetList();
+        var matrix = new[] { list };
+        var f1 = Enumerable.First(list);
+        var f2 = Enumerable.First(GetList());
+        var f3 = Enumerable.First(matrix[0]);
+        Console.WriteLine(Enumerable.First(list));
+    }
+
+    IReadOnlyList<int> GetList()
+    {
+        return new List<int> { 1, 2, 3 };
+    }
+}
+", @"
+using System;
+using System.Linq;
+using System.Collections.Generic;
+class C
+{
+    void M()
+    {
+        var list = GetList();
+        var matrix = new[] { list };
+        var f1 = list[0];
+        var f2 = GetList()[0];
+        var f3 = matrix[0][0];
+        Console.WriteLine(list[0]);
+    }
+
+    IReadOnlyList<int> GetList()
+    {
+        return new List<int> { 1, 2, 3 };
+    }
+}
+", allowNewCompilerDiagnostics: true);
```

Comment:
This should be false everywhere. We don’t prefer code fixes to break existing code.